### DISCUSSION TITLE
Some es indexes not auto-create after upgrade elasticsearch version

### DISCUSSION THREAD

**Discussion by calvin2020** [apache/skywalking#8510] [Q&A]
Hi there:

I encouter a weird question，some es indexes are not auto-created after I upgrade elasticsearch from 6.2.4 to 7.8, can you give me some advices? thank you 
my skywalking version is 8.9
elasticsearch:7.8

![image](https://user-images.githubusercontent.com/27796010/152951288-b6cb4171-ab21-4512-875f-325c207e08b1.png)

**Comment by wu-sheng** [comment]
How did you boot OAP server? Are you running in k8s? If so, there is a `init-container` required in OAP deployment, which is a one-time run.

**Reply by calvin2020** [reply]
> How did you boot OAP server? Are you running in k8s? If so, there is a `init-container` required in OAP deployment, which is a one-time run.

thanks for your reply,I run skywalking in docker container,any other suggestions?

**Reply by wu-sheng** [reply]
Then you need to define what is elasticsearch upgrade. If you are using a clean and empty ElasticSearch 7.x, I would not expect any error. If it is not really new or empty, many some old templates or indices exist.

**Reply by calvin2020** [reply]
> Then you need to define what is elasticsearch upgrade. If you are using a clean and empty ElasticSearch 7.x, I would not expect any error. If it is not really new or empty, many some old templates or indices exist.

I use a clean and empty Elasticsearch 7.x,It has no old templates or indices.after I start skywalking,some indices are auto-created,but some are not,these indices are auto-created as below:
![image](https://user-images.githubusercontent.com/27796010/152963158-8cba586c-82e2-4453-9968-cec705e7be47.png)

**Reply by wu-sheng** [reply]
Do you boot more than one OAP server? Sometimes, concurrency-creating indices could cause this strange case, which is why we provide `init` mode. Besides this, I don't have an idea why some are created and some are not.

**Reply by calvin2020** [reply]
> Do you boot more than one OAP server? Sometimes, concurrency-creating indices could cause this strange case, which is why we provide `init` mode. Besides this, I don't have an idea why some are created and some are not.

No,only one OAP server,I start skywalking via docker-compose,which 'init' parameter should I add in docker-compose.yml?

**Reply by wu-sheng** [reply]
Generally, init mode is already running in default, unless you set booting in no-init mode. But some of indices created, it isn't the issue.

Sorry, you have to debug, I guess. Our ElasticSearch 7 working well, including the demo.skywalking.apache.org.

**Reply by calvin2020** [reply]
> Generally, init mode is already running in default, unless you set booting in no-init mode. But some of indices created, it isn't the issue.
> 
> Sorry, you have to debug, I guess. Our ElasticSearch 7 working well, including the demo.skywalking.apache.org.

OK，thank you ,I'll double-check

**Reply by calvin2020** [reply]
> > Generally, init mode is already running in default, unless you set booting in no-init mode. But some of indices created, it isn't the issue.
> > Sorry, you have to debug, I guess. Our ElasticSearch 7 working well, including the demo.skywalking.apache.org.
> 
> OK，thank you ,I'll double-check

BTW，I have another question,can I   create missing indices manully?

**Reply by wu-sheng** [reply]
You can, but it is on your own, we don't provide scripts or doc for that.

**Reply by calvin2020** [reply]
> You can, but it is on your own, we don't provide scripts or doc for that.

OK,my server's structure is arm64, docker images of OAP server and UI  are available for both x86 and arm64,right?

**Reply by wu-sheng** [reply]
OAP and UI work well in both.

**Reply by calvin2020** [reply]
> OAP and UI work well in both.

I create a index template named sw_profile_task manully,but I still can't start OAP as it can't find the index I create,the error as below:
![image](https://user-images.githubusercontent.com/27796010/153146431-39834bac-350e-410e-ab80-9fb4007420c8.png)

**Reply by wu-sheng** [reply]
I am afraid you have to check the ES about how we request and try through a CURL to see what happens.

**Reply by calvin2020** [reply]
> I am afraid you have to check the ES about how we request and try through a CURL to see what happens.

Nothing special found in ES logs,some requests of adding index template as below:

![image](https://user-images.githubusercontent.com/27796010/153150339-460b6de0-0cfc-4161-bd51-9fa018424e76.png)

**Reply by calvin2020** [reply]
> > I am afraid you have to check the ES about how we request and try through a CURL to see what happens.
> 
> Nothing special found in ES logs,some requests of adding index template as below:
> 
> ![image](https://user-images.githubusercontent.com/27796010/153150339-460b6de0-0cfc-4161-bd51-9fa018424e76.png)

Do you have any other suggestions,I have no idea for this issue currently

**Reply by wu-sheng** [reply]
I don't have one either. Never faced this before.

**Reply by calvin2020** [reply]
> I don't have one either. Never faced this before.

Anyway,thanks for your patient reply
